pkg/resource/table/hooks_replica_updates.go

```diff
@@ -0,0 +1,475 @@
+	a []*v1alpha1.ReplicaGlobalSecondaryIndex,
+	b []*v1alpha1.ReplicaGlobalSecondaryIndex,
+) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	aGSIMap := make(map[string]*v1alpha1.ReplicaGlobalSecondaryIndex)
+	bGSIMap := make(map[string]*v1alpha1.ReplicaGlobalSecondaryIndex)
+
+	for _, gsi := range a {
+		if gsi.IndexName != nil {
+			aGSIMap[*gsi.IndexName] = gsi
+		}
+	}
+
+	for _, gsi := range b {
+		if gsi.IndexName != nil {
+			bGSIMap[*gsi.IndexName] = gsi
+		}
+	}
+
+	for indexName, aGSI := range aGSIMap {
+		bGSI, exists := bGSIMap[indexName]
+		if !exists {
+			return false
+		}
+
+		if !equalReplicaGlobalSecondaryIndexes(aGSI, bGSI) {
+			return false
+		}
+	}
+
+	return true
+}
+
+// equalReplicaArrays returns whether two CreateReplicationGroupMemberAction arrays are equal or not.
+func equalReplicaArrays(a, b []*v1alpha1.CreateReplicationGroupMemberAction) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	aMap := make(map[string]*v1alpha1.CreateReplicationGroupMemberAction)
+	bMap := make(map[string]*v1alpha1.CreateReplicationGroupMemberAction)
+
+	for _, replica := range a {
+		if replica.RegionName != nil {
+			aMap[*replica.RegionName] = replica
+		}
+	}
+
+	for _, replica := range b {
+		if replica.RegionName != nil {
+			bMap[*replica.RegionName] = replica
+		}
+	}
+
+	for regionName, aReplica := range aMap {
+		bReplica, exists := bMap[regionName]
+		if !exists {
+			return false
+		}
+
+		if !equalCreateReplicationGroupMemberActions(aReplica, bReplica) {
+			return false
+		}
+	}
+
+	for regionName := range bMap {
+		if _, exists := aMap[regionName]; !exists {
+			return false
+		}
+	}
+
+	return true
+}
+
+// createReplicaUpdate creates a ReplicationGroupUpdate for creating a new replica
+func createReplicaUpdate(replica *v1alpha1.CreateReplicationGroupMemberAction) svcsdktypes.ReplicationGroupUpdate {
+	replicaUpdate := svcsdktypes.ReplicationGroupUpdate{}
+	createAction := &svcsdktypes.CreateReplicationGroupMemberAction{}
+
+	if replica.RegionName != nil {
+		createAction.RegionName = aws.String(*replica.RegionName)
+	}
+
+	if replica.KMSMasterKeyID != nil {
+		createAction.KMSMasterKeyId = aws.String(*replica.KMSMasterKeyID)
+	}
+
+	if replica.TableClassOverride != nil {
+		createAction.TableClassOverride = svcsdktypes.TableClass(*replica.TableClassOverride)
+	}
+
+	if replica.ProvisionedThroughputOverride != nil {
+		createAction.ProvisionedThroughputOverride = &svcsdktypes.ProvisionedThroughputOverride{}
+		if replica.ProvisionedThroughputOverride.ReadCapacityUnits != nil {
+			createAction.ProvisionedThroughputOverride.ReadCapacityUnits = replica.ProvisionedThroughputOverride.ReadCapacityUnits
+		}
+	}
+
+	if replica.GlobalSecondaryIndexes != nil {
+		gsiList := []svcsdktypes.ReplicaGlobalSecondaryIndex{}
+		for _, gsi := range replica.GlobalSecondaryIndexes {
+			replicaGSI := svcsdktypes.ReplicaGlobalSecondaryIndex{}
+			if gsi.IndexName != nil {
+				replicaGSI.IndexName = aws.String(*gsi.IndexName)
```

Comment:
nit: why not this? or is indexName a different type?
```suggestion
				replicaGSI.IndexName = gsi.IndexName
```